src/Symfony/Component/Notifier/Bridge/Amazon/AmazonTransport.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Amazon;
+
+use AsyncAws\Sns\SnsClient;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Adrien Chinour <github@chinour.fr>
+ *
+ * @experimental in 5.3
+ */
+final class AmazonTransport extends AbstractTransport
+{
+    /** @var SnsClient */
+    private $snsClient;
+
+    public function __construct(SnsClient $snsClient, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->snsClient = $snsClient;
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        $configuration = $this->snsClient->getConfiguration();
+
+        return sprintf('sns://%s?region=%s', $this->getEndpoint(), $configuration->get('region'));
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
```

Comment:
Sns does not only support sms. It also allow sending email, SQS messages or calling a Lambda.

Reply:
Yes, maybe i can add support for email channel, but i don't see how to add others in this notifier bridge.